src/category_theory/monoidal/braided.lean

```diff
@@ -283,4 +298,254 @@ def discrete.braided_functor (F : M →* N) : braided_functor (discrete M) (disc
+                      tensor_μ_def₁,
+                      tensor_comp, tensor_comp] },
+  slice_lhs 3 4 { rw [←tensor_id,
+                      associator_inv_naturality] },
+  slice_lhs 4 5 { rw [←tensor_comp,
+                      associator_naturality,
+                      tensor_comp] },
+  slice_lhs 5 6 { rw [←tensor_comp, ←tensor_comp,
+                      associator_naturality,
+                      tensor_comp, tensor_comp] },
+  slice_lhs 6 10 { rw [←tensor_comp, ←tensor_comp, ←tensor_comp, ←tensor_comp,
+                       ←tensor_comp, ←tensor_comp, ←tensor_comp, ←tensor_comp,
+                       tensor_id,
+                       tensor_associativity_aux,
+                       ←tensor_id,
+                       ←id_comp (𝟙 X₁ ≫ 𝟙 X₁ ≫ 𝟙 X₁ ≫ 𝟙 X₁ ≫ 𝟙 X₁),
+                       ←id_comp (𝟙 Z₂ ≫ 𝟙 Z₂ ≫ 𝟙 Z₂ ≫ 𝟙 Z₂ ≫ 𝟙 Z₂),
+                       tensor_comp, tensor_comp, tensor_comp, tensor_comp, tensor_comp,
+                       tensor_comp, tensor_comp, tensor_comp, tensor_comp, tensor_comp] },
+  slice_lhs 11 12 { rw [←tensor_comp, ←tensor_comp,
+                        iso.hom_inv_id],
+                    simp },
+  simp only [assoc, id_comp],
+  slice_lhs 10 11 { rw [←tensor_comp, ←tensor_comp, ←tensor_comp,
+                        iso.hom_inv_id],
+                    simp },
+  simp only [assoc, id_comp],
+  slice_lhs 9 10 { rw [associator_naturality] },
+  slice_lhs 10 11 { rw [←tensor_comp,
+                        associator_naturality,
+                        tensor_comp] },
+  slice_lhs 11 13 { rw [tensor_id, ←tensor_μ_def₂] },
+  have :
+      ((𝟙 X₁ ⊗ (α_ (X₂ ⊗ Y₁) Z₁ Y₂).inv) ⊗ 𝟙 Z₂) ≫
+      ((𝟙 X₁ ⊗ (α_ X₂ Y₁ Z₁).hom ⊗ 𝟙 Y₂) ⊗ 𝟙 Z₂) ≫
+      (α_ X₁ ((X₂ ⊗ Y₁ ⊗ Z₁) ⊗ Y₂) Z₂).hom ≫
+      (𝟙 X₁ ⊗ (α_ (X₂ ⊗ Y₁ ⊗ Z₁) Y₂ Z₂).hom) ≫
+      (𝟙 X₁ ⊗ (α_ X₂ (Y₁ ⊗ Z₁) (Y₂ ⊗ Z₂)).hom) ≫
+      (α_ X₁ X₂ ((Y₁ ⊗ Z₁) ⊗ Y₂ ⊗ Z₂)).inv
+    = (α_ X₁ ((X₂ ⊗ Y₁) ⊗ (Z₁ ⊗ Y₂)) Z₂).hom ≫
+      (𝟙 X₁ ⊗ (α_ (X₂ ⊗ Y₁) (Z₁ ⊗ Y₂) Z₂).hom) ≫
+      (𝟙 X₁ ⊗ (α_ X₂ Y₁ ((Z₁ ⊗ Y₂) ⊗ Z₂)).hom) ≫
+      (α_ X₁ X₂ (Y₁ ⊗ ((Z₁ ⊗ Y₂) ⊗ Z₂))).inv ≫
+      (𝟙 (X₁ ⊗ X₂) ⊗ (𝟙 Y₁ ⊗ (α_ Z₁ Y₂ Z₂).hom)) ≫
+      (𝟙 (X₁ ⊗ X₂) ⊗ (α_ Y₁ Z₁ (Y₂ ⊗ Z₂)).inv) := by pure_coherence,
+  slice_lhs 7 12 { rw this }, clear this,
+  slice_lhs 6 7 { rw [associator_naturality] },
+  slice_lhs 7 8 { rw [←tensor_comp,
+                      associator_naturality,
+                      tensor_comp] },
+  slice_lhs 8 9 { rw [←tensor_comp,
+                      associator_naturality,
+                      tensor_comp] },
+  slice_lhs 9 10 { rw [associator_inv_naturality] },
+  slice_lhs 10 12 { rw [←tensor_comp, ←tensor_comp,
+                        ←tensor_μ_def₂,
+                        tensor_comp, tensor_comp] },
+  -- It should be possible to close the goal by `coherence` here, but if fails with "Something went
+  -- wrong in the `coherence` tactic: is the target an equation in a monoidal category?",
```

Comment:
Here is something you might wish to debug @semorrison.  (There is a similar example in #13121.)

Reply:
It seems `dsimp, coherence` works fine. I propose we just add a ``try `[dsimp]` in `coherence`.

Reply:
(and I've done so on #13125 now)

Reply:
I see that `dsimp, coherence` works (by the way, with your latest changes I still need to use `dsimp` because `coherence` alone still fails), but I don't understand the effect of `dsimp` that makes `coherence` succeed.  Could you explain?